modules/drivers/athena/src/metabase/driver/athena/hive_parser.clj

```diff
@@ -0,0 +1,62 @@
+(ns metabase.driver.athena.hive-parser
```

Comment:
Don't ask me about this stuff because I didn't write it, Damon did. Unless there is something glaringly unacceptable I would vote towards getting this driver merged and doing cleanup as a follow-on if we need to

Reply:
@camsaul For future context struct/nested fields get returned as strings that we then have to parse in order to get the schema. So this hive parser recursively parses those strings into a schema. 

For example, this
```
array<struct<customerid:string,Order:bigint>>
```

Becomes this
```
[{"customerid": "string", "Order": "bigint"}]
```

Reply:
Ok, we'll have to figure out how to get tests working for that. I turned off nested query support while getting the test suite green. I'll work on re-enabling them and making sure it's tested, maybe as a follow-on